api/client/stats_helpers.go

```diff
@@ -150,7 +154,7 @@ func (s *containerStats) Collect(cli client.APIClient, streamStats bool, waitFir
 				s.mu.Lock()
 				s.err = err
 				s.mu.Unlock()
-				return
+				continue
```

Comment:
we can do it in a different PR if you want, it just seemed like a good thing to do here since in both cases, from the end user perspective, we're dealing with the error case of not getting the data we expected back. up to you...